scanpipe/pipes/d2d.py

```diff
@@ -1109,3 +1111,45 @@ def _map_thirdparty_npm_packages(package_json, to_resources, project):
 
     package_resources.no_status().update(status=flag.NPM_PACKAGE_LOOKUP)
     return package_resources.count()
+
+
+def create_local_files_packages(project):
+    """
+    Create local-files packages for codebase resources not part of a package.
+
+    Resources are grouped by license_expression within a local-files packages.
+    """
+    files_qs = project.codebaseresources.files()
+    to_files_without_package = files_qs.to_codebase().not_in_package()
+    from_files = (
+        files_qs.from_codebase()
+        .has_license_expression()
+        .filter(
+            related_to__to_resource__in=Subquery(to_files_without_package.values("pk"))
+        )
+    )
+
+    # Do not include any other fields in the ``values()``
+    grouped_by_license = from_files.values("detected_license_expression").order_by(
+        "detected_license_expression"
+    )
+
+    grouped_by_license = grouped_by_license.annotate(
+        grouped_resource_ids=ArrayAgg("id", distinct=True),
+        grouped_copyrights=ArrayAgg("copyrights", distinct=True),
+    )
+
+    for group in grouped_by_license:
+        codebase_resource_ids = set(group["grouped_resource_ids"])
+        copyrights = set(
```

Comment:
IMHO we should use a `Counter` instead and then report copyrights sorted in order of most frequent down. This could be a second step though

Reply:
Counter implemented, thanks for the suggestion.